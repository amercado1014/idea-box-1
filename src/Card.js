import React from 'react';

const Card = ({title, description, removeIdea, id}) => (
  <div>
    <h2>{title}</h2>
    <p>{description}</p>
    <button onClick={() => removeIdea(id)}
            className='remove-idea'
      >Remove</button>
  </div>
)

export default Card
