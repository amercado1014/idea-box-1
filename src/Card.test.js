import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'

describe('Card Tests', () => {
  it('should match the yung-snapshot', () => {
    const idea = { id: 1, title: 'totes full', description: 'i ate alot of food' }
    const mockFunc = jest.fn()
    const { id, title, description } = idea
    const renderedApp = shallow(<Card id={id}
                                      title={title}
                                      description={description}
                                      removeIdea={mockFunc}
                                />)
    const removeButton = renderedApp.find('.remove-idea')
    const calls = 1
    removeButton.simulate('click')

    expect(mockFunc.mock.calls.length).toEqual(calls);
    expect(renderedApp).toMatchSnapshot()
  })
});
