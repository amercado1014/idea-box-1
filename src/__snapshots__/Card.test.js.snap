// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Tests should match the yung-snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    description="i ate alot of food"
    id={1}
    removeIdea={[Function]}
    title="totes full"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          totes full
</h2>,
        <p>
          i ate alot of food
</p>,
        <button
          className="remove-idea"
          onClick={[Function]}
>
          Remove
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "totes full",
        },
        "ref": null,
        "rendered": "totes full",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "i ate alot of food",
        },
        "ref": null,
        "rendered": "i ate alot of food",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove",
          "className": "remove-idea",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            totes full
</h2>,
          <p>
            i ate alot of food
</p>,
          <button
            className="remove-idea"
            onClick={[Function]}
>
            Remove
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "totes full",
          },
          "ref": null,
          "rendered": "totes full",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "i ate alot of food",
          },
          "ref": null,
          "rendered": "i ate alot of food",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove",
            "className": "remove-idea",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
